analyses/immune-deconv/02-visualize_quantiseq.Rmd

```diff
@@ -260,30 +270,112 @@ ggplot(pdl1_cd8) +
 
 Instead, let's explore the PDL1 distributions specifically and export this figure:
 
-```{r pdl1_distributions, fig.width = 8, fig.height = 8}
+```{r pdl1_distributions, fig.width = 7, fig.height = 5}
 
 # distributions of PDL1 expression
 pdl1_distributions_plot <- ggplot(pdl1_cd8) + 
-  aes(x = fct_reorder(molecular_subtype,log2_expression, .desc=T),
+  #aes(x = fct_reorder(molecular_subtype,log2_expression, .desc=T),
+  aes(x = molecular_subtype,
       y = log2_expression) +
   # remove outliers
   geom_boxplot(outlier.shape = NA, color = "grey40", size = 0.2) + 
   geom_jitter(width = 0.15, size = 0.75, alpha = 0.8, aes(color = cancer_group_hex)) + 
-  facet_wrap(~cancer_group_display, nrow = 2, scales = "free") +
+  #facet_wrap(~cancer_group_display, nrow = 2, scales = "free") +
   scale_color_identity() +
   labs(x = "Molecular subtype of sample",
-       y = "Log2 expression (FPKM) of PDL1") +
+       y = "Log2 expression (FPKM) of CD274") +
   ggpubr::theme_pubr() + 
   theme(axis.text.x = element_text(hjust = 1, 
                                    size = rel(0.6), 
-                                   angle = 90)) 
+                                   angle = 75)) 
 
 pdl1_distributions_plot
 
-ggsave(pdl1_distributions_plot_file, pdl1_distributions_plot, width = 8, height = 8)
+ggsave(pdl1_distributions_plot_file, pdl1_distributions_plot, width = 7, height = 5)
+
+```
+Visualize MB only, since that was the only significant result from the survival analyses
+
+```{r pdl1_mb, fig.width = 5, fig.height = 5}
+# select only mb samples
+pdl1_cd8_mb <- pdl1_cd8 %>%
+  filter(cancer_group_display == "Medulloblastoma")
+
+
+# Count group sizes to use in x-axis labels. Use this data frame going forward
+  df_counts <- pdl1_cd8_mb %>%
+    group_by(molecular_subtype) %>%
+    mutate(subtype_count = n()) %>%
+    ungroup() %>%
+    mutate(molecular_subtype = glue::glue("{molecular_subtype}\n(N = {subtype_count})"))
+  
+  wilcox_df <- ggpubr::compare_means(log2_expression ~ molecular_subtype, 
+                             data = df_counts,
+                             method = "wilcox.test")
+  
+# distributions of PDL1 expression
+cd274_plot_mb <- ggplot(pdl1_cd8_mb) + 
+  aes(x = molecular_subtype,
+      y = log2_expression) +
+  # remove outliers
+  geom_boxplot(outlier.shape = NA, color = "grey40", size = 0.2) + 
+  geom_jitter(width = 0.15, size = 0.75, alpha = 0.8, aes(color = "black")) + 
+  scale_color_identity() +
+  labs(x = "Molecular subtype of sample",
+       y = "Log2 expression (FPKM) of CD274") +
+  ggpubr::theme_pubr() + 
+  theme(axis.text.x = element_text(hjust = 1, 
+                                   size = rel(0.6), 
+                                   angle = 75)) 
```

Comment:
For getting the P-values in the plot, see these docs and in particular the example: https://rpkgs.datanovia.com/ggpubr/reference/stat_pvalue_manual.html

You need to add a column to `wilcox_df` called `y.position` that contains the (manually determined!) y-axis coordinates for where each label will be placed.